napari/plugins/_builtins.py

```diff
@@ -29,3 +37,138 @@ def napari_get_reader(path: Union[str, List[str]]) -> ReaderFunction:
         function that returns layer_data to be handed to viewer._add_layer_data
     """
     return image_reader_to_layerdata_reader(magic_imread)
+
+
+@napari_hook_implementation(trylast=True)
+def napari_write_image(path: str, data: Any, meta: dict) -> bool:
+    """Our internal fallback image writer at the end of the plugin chain.
+
+    Parameters
+    ----------
+    path : str
+        Path to file, directory, or resource (like a URL).
+    data : array or list of array
+        Image data. Can be N dimensional. If meta['rgb'] is `True` then the
+        data should be interpreted as RGB or RGBA. If meta['is_pyramid'] is
+        True, then the data should be interpreted as an image pyramid.
+    meta : dict
+        Image metadata.
+
+    Returns
+    -------
+    bool : Return True if data is successfully written.
+    """
+    imsave(path, data)
+    return True
+
+
+@napari_hook_implementation(trylast=True)
+def napari_write_points(path: str, data: Any, meta: dict) -> bool:
+    """Our internal fallback points writer at the end of the plugin chain.
+
+    Append `.csv` extension to the filename if it is not already there.
+
+    Parameters
+    ----------
+    path : str
+        Path to file, directory, or resource (like a URL).
+    data : array (N, D)
+        Coordinates for N points in D dimensions.
+    meta : dict
+        Points metadata.
+
+    Returns
+    -------
+    bool : Return True if data is successfully written.
+    """
+    ext = os.path.splitext(path)[1]
+    if ext == '':
+        path = path + '.csv'
+    elif ext != '.csv':
+        raise ValueError('If an extension if provided then it must be `.csv`')
+
+    # construct table from data
```

Comment:
This has now been improved - the `examples/add_points_with_properties.py` now looks as followed when loaded into other software
![image](https://user-images.githubusercontent.com/6531703/79140205-76b90b80-7d6c-11ea-80ae-f18f900405fe.png)

I went with `index` and `axis-` as those words are intuitive for people not familiar with napari. Thoughts @jni @GenevieveBuckley @kevinyamauchi

Reply:
I love it!

Reply:
Me too!